File: src/android/db/sqlcipher.kt
package com.calisto.vault.db

import net.sqlcipher.database.SQLiteDatabase
import net.sqlcipher.database.SQLiteOpenHelper
import android.content.Context

class VaultDB(context: Context) : SQLiteOpenHelper(context, "vault.db", null, 1) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("""
            CREATE TABLE IF NOT EXISTS entries (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                title TEXT,
                username TEXT,
                password TEXT,
                notes TEXT,
                created_at TEXT
            )
        """)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {}

    fun open(): SQLiteDatabase {
        SQLiteDatabase.loadLibs(context)
        return getWritableDatabase("calisto_master_key")
    }
}

File: src/windows/db/sqlcipher.js
const path = require('path');
const Database = require('better-sqlite3');

const dbPath = path.join(__dirname, 'vault.db');

function initDatabase() {
    const db = new Database(dbPath);
    db.pragma("key = 'calisto_master_key'");
    db.prepare(`
        CREATE TABLE IF NOT EXISTS entries (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            title TEXT,
            username TEXT,
            password TEXT,
            notes TEXT,
            created_at TEXT
        )
    `).run();
    return db;
}

module.exports = { initDatabase };
